src/test/java/io/supertokens/test/GetUsersByEmailAPITest2_8.java

```diff
@@ -0,0 +1,150 @@
+            return;
+        }
+
+        // given
+        {
+            JsonObject signUpResponse = Utils.signInUpRequest_2_7(process, "john.doe@example.com", true, "mockThirdParty", "mockThirdPartyUserId");
+            assertEquals(signUpResponse.get("status").getAsString(), "OK");
+
+            JsonObject signUpUser = signUpResponse.get("user").getAsJsonObject();
+            assertEquals(signUpUser.get("email").getAsString(), "john.doe@example.com");
+            assertNotNull(signUpUser.get("id"));
+        }
+        {
+            JsonObject signUpResponse = Utils.signInUpRequest_2_7(process, "john.doe@example.com", true, "mockThirdParty2", "mockThirdParty2UserId");
+            assertEquals(signUpResponse.get("status").getAsString(), "OK");
+
+            JsonObject signUpUser = signUpResponse.get("user").getAsJsonObject();
+            assertEquals(signUpUser.get("email").getAsString(), "john.doe@example.com");
+            assertNotNull(signUpUser.get("id"));
+        }
+
+        // when
+        HashMap<String, String> query = new HashMap<>();
+        query.put("email", "john.doe@example.com");
+
+        JsonObject response = io.supertokens.test.httpRequest.HttpRequest
+                .sendGETRequest(process.getProcess(), "",
+                        "http://localhost:3567/recipe/users/by-email", query, 1000,
+                        1000,
+                        null, "2.8", RECIPE_ID.THIRD_PARTY.toString());
+
+        // then
+        JsonArray jsonUsers = response.get("users").getAsJsonArray();
+
+        jsonUsers.forEach(jsonUser -> assertEquals("john.doe@example.com", jsonUser.getAsJsonObject().get("email").getAsString()));
+
+        assertEquals("OK", response.get("status").getAsString());
+        assertEquals(2, response.entrySet().size());
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testShouldThrowOnBadInput() throws Exception {
+        // setup
+        String[] args = {"../"};
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        // when
+        try {
+            Map<String, String> emptyQuery = new HashMap<>();
+
+            testBadInput(process, emptyQuery);
+        // then
+        } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+            assertEquals(400, e.statusCode);
+            assertEquals("Http error. Status Code: 400. Message: email cannot be empty", e.getMessage());
+        }
+
+        // when
+        try {
+            Map<String, String> invalidQuery = new HashMap<>();
+
+            invalidQuery.put("email", "");
```

Comment:
Since you are checking for email regex validation, you should also add that check?

Reply:
It's an edge case. If I didn't test that here, I would anyway test regex if it matched an empty string.